app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -756,37 +756,50 @@ void OpenXRInputSource::Update(const XrFrameState& frameState, XrSpace localSpac
     UpdateHaptics(delegate);
 }
 
-XrResult OpenXRInputSource::UpdateInteractionProfile(ControllerDelegate& delegate)
+XrResult OpenXRInputSource::UpdateInteractionProfile(ControllerDelegate& delegate, const char *emulateProfile)
 {
-    XrInteractionProfileState state { XR_TYPE_INTERACTION_PROFILE_STATE };
-    RETURN_IF_XR_FAILED(xrGetCurrentInteractionProfile(mSession, mSubactionPath, &state));
-    if (state.interactionProfile == XR_NULL_PATH) {
-      return XR_SUCCESS; // Not ready yet
-    }
+    const char *path = nullptr;
+    size_t path_len = 0;
+
+    if (emulateProfile == nullptr) {
+        XrInteractionProfileState state{XR_TYPE_INTERACTION_PROFILE_STATE};
+        RETURN_IF_XR_FAILED(xrGetCurrentInteractionProfile(mSession, mSubactionPath, &state));
+        if (state.interactionProfile == XR_NULL_PATH) {
+            return XR_SUCCESS; // Not ready yet
+        }
 
-    constexpr uint32_t bufferSize = 100;
-    char buffer[bufferSize];
-    uint32_t writtenCount = 0;
-    RETURN_IF_XR_FAILED(xrPathToString(mInstance, state.interactionProfile, bufferSize, &writtenCount, buffer));
+        constexpr uint32_t bufferSize = 100;
+        char buffer[bufferSize];
+        uint32_t writtenCount = 0;
+        RETURN_IF_XR_FAILED(xrPathToString(mInstance, state.interactionProfile,
+                                           bufferSize, &writtenCount,
+                                           buffer));
+        path = buffer;
+        path_len = writtenCount;
+    } else {
+        path = emulateProfile;
+        path_len = strlen(emulateProfile);
+    }
 
     mActiveMapping = nullptr;
 
     for (auto& mapping : mMappings) {
-        if (!strncmp(mapping.path, buffer, writtenCount)) {
+        if (!strncmp(mapping.path, path, path_len)) {
             mActiveMapping = &mapping;
             break;
         }
     }
 
+    if (!mActiveMapping)
+        return XR_ERROR_PATH_INVALID;
+
     // Add haptic devices to controller, if any
-    if (mActiveMapping != nullptr) {
-        uint32_t numHaptics = 0;
-        for (auto& haptic: mActiveMapping->haptics) {
-            if (haptic.hand == OpenXRHandFlags::Both || haptic.hand == mHandeness)
-                numHaptics++;
-        }
-        delegate.SetHapticCount(mIndex, numHaptics);
+    uint32_t numHaptics = 0;
+    for (auto& haptic: mActiveMapping->haptics) {
+        if (haptic.hand == OpenXRHandFlags::Both || haptic.hand == mHandeness)
+            numHaptics++;
     }
+    delegate.SetHapticCount(mIndex, numHaptics);
```

Comment:
These changes are nice but as they are not needed I think we can leave them for a followup.

Reply:
Ok, I reverted them.

I changed  that because I realized I was not checking that mActiveMapping has to be non-null for button count logic. And there was a check above already, so.